public/app/features/browse-dashboards/api/browseDashboardsAPI.ts

```diff
@@ -36,21 +52,188 @@ export const browseDashboardsAPI = createApi({
+    }),
+    moveFolder: builder.mutation<void, { folder: FolderDTO; destinationUID: string }>({
+      invalidatesTags: ['getFolder'],
+      query: ({ folder, destinationUID }) => ({
+        url: `/folders/${folder.uid}/move`,
+        method: 'POST',
+        data: { parentUID: destinationUID },
+      }),
+      onQueryStarted: ({ folder, destinationUID }, { queryFulfilled, dispatch }) => {
+        const { parentUid } = folder;
+        queryFulfilled.then(() => {
+          dispatch(
+            refetchChildren({
+              parentUID: parentUid,
+              pageSize: parentUid ? PAGE_SIZE : ROOT_PAGE_SIZE,
+            })
+          );
+          dispatch(
+            refetchChildren({
+              parentUID: destinationUID,
+              pageSize: destinationUID ? PAGE_SIZE : ROOT_PAGE_SIZE,
+            })
+          );
+        });
+      },
+    }),
+    newFolder: builder.mutation<FolderDTO, { title: string; parentUid?: string }>({
+      query: ({ title, parentUid }) => ({
+        method: 'POST',
+        url: '/folders',
+        data: {
+          title,
+          parentUid,
+        },
+      }),
+      onQueryStarted: ({ parentUid }, { queryFulfilled, dispatch }) => {
+        queryFulfilled.then(async ({ data: folder }) => {
+          await contextSrv.fetchUserPermissions();
+          dispatch(notifyApp(createSuccessNotification('Folder created')));
+          dispatch(
+            refetchChildren({
+              parentUID: parentUid,
+              pageSize: parentUid ? PAGE_SIZE : ROOT_PAGE_SIZE,
+            })
+          );
+          locationService.push(locationUtil.stripBaseFromUrl(folder.url));
+        });
+      },
     }),
     saveFolder: builder.mutation<FolderDTO, FolderDTO>({
-      invalidatesTags: (_result, _error, args) => [{ type: 'getFolder', id: args.uid }],
-      query: (folder) => ({
+      // because the getFolder calls contain the parents, renaming a parent/grandparent/etc needs to invalidate all child folders
+      // we could do something smart and recursively invalidate these child folders but it doesn't seem worth it
+      // instead let's just invalidate all the getFolder calls
+      invalidatesTags: ['getFolder'],
+      query: ({ uid, title, version }) => ({
         method: 'PUT',
-        showErrorAlert: false,
-        url: `/folders/${folder.uid}`,
+        url: `/folders/${uid}`,
         data: {
-          title: folder.title,
-          version: folder.version,
+          title,
+          version,
         },
       }),
+      onQueryStarted: ({ parentUid }, { queryFulfilled, dispatch }) => {
+        queryFulfilled.then(() => {
+          dispatch(
+            refetchChildren({
+              parentUID: parentUid,
+              pageSize: parentUid ? PAGE_SIZE : ROOT_PAGE_SIZE,
+            })
+          );
+        });
+      },
+    }),
+    deleteItems: builder.mutation<void, DeleteItemsArgs>({
```

Comment:
might be a good idea to organise these funtions it's logical groups, and add like a 1 line commente above each explaining what they're for. Something like:

 - getFolder
 - newFolder
 - saveFolder
 - moveFolder
 - deleteFolder

 - getAffectedItems
 - moveItems
 - deleteItems

 - saveDashboard

Looking at this now it took me a bit to remember what the *Items methods are for